pysnooper/pycompat.py

```diff
@@ -80,3 +80,19 @@ def time_isoformat(time, timespec='microseconds'):
         return result
 
 
+if sys.version_info[:2] >= (3, 7):
+    time_fromisoformat = datetime_module.time.fromisoformat
+else:
+    def time_fromisoformat(isoformat_str):
+        hour, minute, second, microsecond = map(
+            int,
+            isoformat_str.replace('.', ':').split(':'))
+        return datetime_module.time(hour, minute, second, microsecond)
+
+
+def timedelta_isoformat(timedelta, timespec='microseconds'):
+    assert isinstance(timedelta, datetime_module.timedelta)
+    if timespec != 'microseconds':
+        raise NotImplementedError
```

Comment:
```suggestion
def timedelta_isoformat(timedelta):
```

There's no need to handle `timespec` as this doesn't need to be compatible with an existing method. And the line below will fail nicely if a non-timedelta is passed.

Reply:
Why do you suggest deleting the assertion of checking datetime_module.timedelta?

Reply:
It's just noise in the code, and as I said the type is enforced by the rest of the code anyway. If you call `timedelta_isoformat(3)`, without the assertion you immediately get an error `TypeError: unsupported operand type(s) for +: 'datetime.datetime' and 'int'`. With the assertion, you get:

```
    assert isinstance(timedelta, datetime_module.timedelta)
AssertionError
```

which is actually less informative.

Reply:
I see. I agree with your idea.